### DISCUSSION TITLE
Node resize handles get stuck on palette tool

### DISCUSSION THREAD

**Discussion by jmenzies12** [eclipse-glsp/glsp#1448] [Q&A]
Hey there,

I noticed some odd behavior with the resize node handles. If you drag one of the handles to resize the bounds of a node, and the handle touches the palette, let go of the mouse button and then click back on the handle, the node gets left at that size and does not trigger a change bounds request.

![image](https://github.com/user-attachments/assets/e9765419-1599-4a48-b2a0-6a67f289efe4)

This is in the glsp-vscode-integration. It takes some effort to reproduce so it's nothing crazy, just wanted to point it out.

Cheers
Jonathan

**Comment by tortmayr** [comment]
Hi @jmenzies12,
I think this was fixed recently https://github.com/eclipse-glsp/glsp-client/pull/399.
(@martin-fleck-at Can you confirm that this should be fixed?)
The change will be part of the upcoming 2.3 release which is planned for this week.

**Reply by jmenzies12** [reply]
Hey @tortmayr,

We recently upgraded to the 2.3 release and can still reproduce the issue with the steps mentioned above, if you need any more information please let me know.

Cheers
Jonathan

**Reply by tortmayr** [reply]
@jmenzies12 

Maybe I did not clarify this enough. The issue that has been fixed is that you can continue the resize operation when moving back from the toolbox to the diagram:
https://github.com/user-attachments/assets/645be5e0-3343-4580-905c-e836346e9d72

However, the resize operation will still stuck or pause when the mouse is over the tool palette.
From the top of my head I can think of three potential solutions for this:
- Disable pointer event for the tool palette while a resize or move is in progress. This would allow the resize animation to continue even if the mouse is technically over the tool palette. However, since the resizing is happening on the svg which is overlayed by the tool palette most of this resizing would not be visible and to the user it most likely would still appear as if the operation is stuck
- Shift the diagram canvas whenever the mouse would reach the tool palette. While this would ensure a smooth resize operation shifting the canvas still might be a bit unexpected/unconventional.
- Alternatively the tool palette could also be hidden during a move/resize operation. This approach would probably work the best but is something that should not be implemented by default. It should be an opt-in behavior configurable by the adopter.

Please let me know if you have any better solution or ideas. From my point of view all of these approaches are not ideal and as you already mentioned this is a rather low-priority issue. So maybe we should just keep the current behavior for now.

**Reply by jmenzies12** [reply]
Hey @tortmayr 

Just to clarify, the sticking/pausing on the palette is understandable and the fix makes sense for that.

The issue that I am having is, if the resize tool gets stuck on the palette and you let go of the mouse button (or trackpad button), when you go back to the canvas the resize tool will let you continue, however when you finally click to save the resizing it doesn't trigger the change bounds handler to apply the change (in my case the node is resized to keep it's shape when the change bounds is applied, which does not happen if the changes are not applied). I've included a video below to help explain what I mean.

https://github.com/user-attachments/assets/71cf6160-f48e-4f33-a82b-8d8702ab0ef5

Let me know if you need any more information.

Cheers
Jonathan
